### PR TITLE
CP-22306: Refactor release notes process

### PR DESCRIPTION
**Pull request by josephbarnett**
### Description

1. Add PM as codeowner
2. Add `docs/releases/` as release doc origin
3. Add verify release note file step
4. Add manual approval process for release

 
### References

* [Slack](https://cloudzero.slack.com/archives/C06NXMJC9PU/p1727883714420999?thread_ts=1727883670.245249&cid=C06NXMJC9PU)

### CODE REVIEW

**Review by conradcz — COMMENTED** [review]
Looks good to me!

File: docs/releases/RELEASE_PROCESS.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+---
+
+## Creating Release Notes
+
+Effective release notes provide clear and concise information about the changes in each release. Follow these guidelines to create comprehensive release notes.
+
+### Pro-Tip
+
+- **Review Changes:** Examine the GitHub diff between the previous and current release to identify all changes.
+- **Commit Messages:** Use commit titles to outline the changes and categorize them appropriately.
```
Comment (beckilee) [inline docs/releases/RELEASE_PROCESS.md:84]:
```suggestion
- **Commit Messages:** Use commit titles to outline the changes and categorize them appropriately. Include a link to the relevant commit where possible.
```

File: docs/releases/RELEASE_PROCESS.md [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+- **Content:**
+  - Minor updates or maintenance tasks.
+  - Documentation updates.
+- **Note:** Aim to keep this section empty by categorizing changes appropriately.
+
+---
+
+### Release Notes Template
+
+Copy and paste the following template to create your release notes. Replace placeholders with relevant information.
```
Comment (beckilee) [inline docs/releases/RELEASE_PROCESS.md:150]:
```suggestion
Copy and paste the following template to create your release notes. Replace placeholders with relevant information. Remove any headings you don't need.
```

**Review by beckilee — APPROVED** [review]
Just two small suggestions on the process doc -- feel free to take them or leave them 🙂 Thank you for this -- it looks fantastic! 🤩

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -1 +1 @@
-*       @cloudzero/cirrus
+*       @cloudzero/cirrus
```
Comment (roberthocking) [inline .github/CODEOWNERS:1]: Weren't we going to add PM here too, or is that in another rep?

Comment (josephbarnett) [reply .github/CODEOWNERS:1]: No need. That is part of the `deployment environment` configuration which is where the signoff is collected. See here: https://github.com/Cloudzero/cloudzero-charts/settings/environments

**Review by dmepham — APPROVED** [review]
lgtm! does the process change for beta releases at all?

**Comment by josephbarnett** [comment]
> lgtm! does the process change for beta releases at all?

Yes - this affects both processes. There is magic in that little `environment: release`  addition to those two workflows. ☺️

**Review by wreckedred — APPROVED** [review]
Thank you - lgtm

**Review by conradcz — APPROVED** [review]

**Pull request merged**